### PR TITLE
EES-3192: Add error page redirection to Notifier on verification failure

### PR DESCRIPTION
**Pull request by tomjonesdev**
Public website users subscribing to publication notifications were presented with a generic, non-user-friendly error message if they attempted to verify their email address after the verification token had expired.

This change adds an error page matching existing error page format and style (and meeting the GDS standards) to inform users of the issue, and how to resolve it.

image

### CODE REVIEW

File: src/explore-education-statistics-frontend/src/modules/subscriptions/VerificationErrorPage.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { NextPage } from 'next';
+import React from 'react';
+
+const VerificationErrorPage: NextPage = () => {
+  return (
+    <Page title="Verification failed">
+      <div className="govuk-panel__body">
+        <p>
+          Your subscription verification token has expired. You can try again by
+          re-subscribing from the publication's main screen.
```
Comment (sambiramairelogic) [inline src/explore-education-statistics-frontend/src/modules/subscriptions/VerificationErrorPage.tsx:11]: This may be scope creep, but it might be a nicer user journey if we were to make `publication's main screen` a link to said main screen. Could we send the publication id in the RedirectResult and add it as a prop here?

File: src/GovUk.Education.ExploreEducationStatistics.Notifier/Functions/SubscriptionManager.cs [hunk 2]
```diff
@@ -254,7 +254,8 @@ public class SubscriptionManager
                 }
             }
 
-            return new BadRequestObjectResult("Unable to verify subscription");
+            return new RedirectResult(
+                $"{_appSettingOptions.PublicAppUrl}/subscriptions/verification-error");
```
Comment (sambiramairelogic) [inline src/GovUk.Education.ExploreEducationStatistics.Notifier/Functions/SubscriptionManager.cs:258]: I'm not sure it's correct to use a redirect here - what's actually occurred? If the token has expired shouldn't we get a 403 or 401? Or if it's a bad payload maybe a 422? I'd prefer to return a 4xx status code and handle the problem in the frontend server.

Using a redirect will also have SEO implications - firstly we'd need to make sure it's a permeant not a temporary redirect, and secondly we'd then need to stop Google indexing the error verification page as the canonical. Perfectly doable, but feels like there are better options than using a redirect maybe?

File: src/explore-education-statistics-frontend/src/modules/subscriptions/VerificationErrorPage.tsx (hunk 1 above)
Comment (tomjonesdev) [reply src/explore-education-statistics-frontend/src/modules/subscriptions/VerificationErrorPage.tsx:11]: I considered this, but the data isn't available at this stage because the token linking a user to a publication has been deleted. Mentioned to Cam and he's happy with current messaging, and can iterate down the line if it's deemed worth the effort investigating.

File: src/GovUk.Education.ExploreEducationStatistics.Notifier/Functions/SubscriptionManager.cs (hunk 2 above)
Comment (tomjonesdev) [reply src/GovUk.Education.ExploreEducationStatistics.Notifier/Functions/SubscriptionManager.cs:258]: It was a 400. The issue was that the Notifier is a separate endpoint unrelated to the Public site on which we wanted to display the error page, so it can't be handled by middleware. `RedirectResult` is used in other partS of this class for subscription update/unsubscribe.
